packages/esm-patient-chart-app/src/workspace/workspace-window.component.tsx

```diff
@@ -44,6 +59,44 @@ const WorkspaceWindow: React.FC<ContextWorkspaceParams> = () => {
     closeWorkspace = () => {},
   } = useMemo(() => workspaces?.[0] ?? ({} as OpenWorkspace), [workspaces]);
 
+  useEffect(() => {
+    const handleRouting = (event) => {
+      const {
+        detail: { cancelNavigation, newUrl },
+      } = event as { detail: { cancelNavigation: () => void; newUrl: string } };
+
+      const regex = new RegExp(/\/patient\/([a-zA-Z0-9\-]+)\/?/);
+      const isNewUrlPatientChartUrl = regex.test(newUrl);
+      const canCloseAllWorkspaces = getWorkspaceStore()
+        .getState()
+        .openWorkspaces.every(({ name }) => {
+          const canCloseWorkspace = getWhetherWorkspaceCanBeClosed(name);
+          return canCloseWorkspace;
+        });
+
+      if (!isNewUrlPatientChartUrl) {
+        if (!canCloseAllWorkspaces) {
+          cancelNavigation();
+          const navigateToNewUrl = () => {
+            function getUrlWithoutPrefix(url: string) {
+              return url.split(window['getOpenmrsSpaBase']())?.[1];
+            }
+            navigate({ to: `\${openmrsSpaBase}/${getUrlWithoutPrefix(newUrl)}` });
+          };
+
+          closeAllWorkspaces(navigateToNewUrl);
+        } else {
+          resetWorkspaceStore();
```

Comment:
You should not be using `resetWorkspaceStore`, per the docstring for that function. What is your intention? Why should the workspace state be cleared?

Reply:
Since the user is leaving the patient chart, the workspace can be reset to the initial state.